server/controllers/items/lib/validate_item_async.js

```diff
@@ -2,6 +2,7 @@ const _ = require('builders/utils')
 const error_ = require('lib/error/error')
 const { validateVisibilityKeys } = require('lib/visibility/visibility')
 const getEntitiesByUris = require('controllers/entities/lib/get_entities_by_uris')
+const { flatMapUniq, mapUniq } = require('lib/utils/base')
```

Comment:
i thought the convention was to extract only if used by several consumers.
I would prefer to keep a expression such as `_.uniq(_.map(items, 'shelves').flat())` which is pretty straight forward to me

Reply:
several reasons push me toward preferring using the extract syntax by default:
- that's preparing the ground for the transition to the ES module syntax (already used on the client): those extracted variables will simply become named exports (ex: `export function flatMapUniq`), rather than key/values in a `export default` object
- there are cases where keeping a reference to an extracted variable rather than the object that contained it allows that object (and the other things it keeps a reference to) to be garbage collected, freeing memory. That would not be the case with `lib/utils/base`, as that object is anyway referenced all over the code base, but I think it's a good reason to have it as a default, allowing to not think to much about it
- my understanding is that there is a small performance benefit to do it like this: calling `_.uniq` rather than directly `uniq` means having to perform a lookup on the `_` object. The cost for that performance benefit being that we keep a reference to all those functions (`uniq`, `map`, etc) rather than just to their parent object (`_`)
- also, the auto-import in vscodium makes it easier to import like this